### DISCUSSION TITLE
Generate MacOS installer

### DISCUSSION THREAD

**Discussion by troian** [goreleaser/goreleaser#1413] [Ideas, Issue Triage, and General discussions]
I know there were a few issues with the same headline but lots of time went since.
Is there any option to generate **pkg** installer

**Comment by caarlos0** [comment]
Hey, no there isn't right now.

you can check a couple of related issues:

- https://github.com/goreleaser/goreleaser/issues/1227
- https://github.com/goreleaser/goreleaser/issues/744

you could in theory use the sign pipe to generate the pkg... but this will likely not be added to goreleaser unless there is a way to do it on all platforms...

**Reply by troian** [reply]
Cheers

**Reply by jrschumacher** [reply]
@caarlos0 is there a way to chain signing? Since the `.app` needs to be signed first it seems that if I wanted to support DMG and pkg I'd need a duplicate flow in the `signs` pipe to get the desired output.

**Reply by caarlos0** [reply]
its already supported: https://goreleaser.com/customization/notarize/#native

**Reply by jrschumacher** [reply]
Thanks, this is super helpful. So I can use notarize to sign the appbundle first then have the hacky `pkgbuild` / `productbuild` in the sign step and build that (then sign it in the same step)?

**Reply by caarlos0** [reply]
right now it only works if you also package it as a dmg - not sure if you can hack around it with pkgbuild still 

making the pkg stuff is on my todo list btw

**Reply by jrschumacher** [reply]
That's fair. I don't see any reason why I wouldn't want to build a DMG and a pkg. DMGs are useful for end-users who control their own machines (and if I don't need to run a pre/post scripts) while mobile device managers (MDM) seem to want PKGs.

**Comment by caarlos0** [comment]
FWIW this has been added:

- https://goreleaser.com/customization/dmg/
- https://goreleaser.com/customization/app_bundles/
- https://goreleaser.com/customization/notarize/
